pmd-java/src/main/java/net/sourceforge/pmd/lang/java/rule/performance/UseStringBufferForStringAppendsRule.java

```diff
@@ -39,81 +31,31 @@ public UseStringBufferForStringAppendsRule() {
                 // ignore if the field or formal parameter is *not* used within loops
                 continue;
             }
-            ASTConditionalExpression conditional = name.getFirstParentOfType(ASTConditionalExpression.class);
 
-            if (conditional != null) {
-                Node thirdParent = name.getNthParent(3);
-                Node fourthParent = name.getNthParent(4);
-                if ((Objects.equals(thirdParent, conditional) || Objects.equals(fourthParent, conditional))
-                        && conditional.getFirstParentOfType(ASTStatementExpression.class) == statement) {
-                    // is used in ternary as only option (not appended to other
-                    // string)
-                    continue;
-                }
-            }
-            if (statement.getNumChildren() > 0 && statement.getChild(0) instanceof ASTPrimaryExpression) {
-                ASTName astName = statement.getChild(0).getFirstDescendantOfType(ASTName.class);
-                if (astName != null) {
-                    ASTAssignmentOperator assignmentOperator = statement
-                            .getFirstDescendantOfType(ASTAssignmentOperator.class);
-                    if (astName.equals(name)) {
-                        if (assignmentOperator != null && assignmentOperator.isCompound()) {
-                            if (isWithinLoop(name)) {
-                                // always report within a loop
-                                addViolation(data, assignmentOperator);
-                            } else {
-                                usageCounter++;
-                                if (usageCounter > 1) {
-                                    // only report, if it is not the first time
-                                    addViolation(data, assignmentOperator);
-                                }
-                            }
-                        }
-                    } else if (astName.hasImageEqualTo(name.getImage())) {
-                        if (assignmentOperator != null && !assignmentOperator.isCompound()) {
-                            if (isWithinLoop(name)) {
-                                // always report within a loop
-                                addViolation(data, assignmentOperator);
-                            } else {
-                                usageCounter++;
-                                if (usageCounter > 1) {
-                                    // only report, if it is not the first time
-                                    addViolation(data, assignmentOperator);
-                                }
-                            }
-                        } else if (assignmentOperator != null && assignmentOperator.isCompound()
-                                && usageCounter >= 1) {
-                            addViolation(data, assignmentOperator);
-                        }
+            if (usage.getAccessType() == AccessType.WRITE) {
```

Comment:
I think you should here also check that the assignment is compound OR it's `=` but some term of the concatenation is the variable.

Otherwise you get FPs like this:

https://github.com/checkstyle/checkstyle/blob/checkstyle-8.10/src/it/java/com/google/checkstyle/test/base/AbstractModuleTestSupport.java#L310